docs/using-hive.md

```diff
@@ -2,72 +2,231 @@
+
+### Pull Secret
+
+OpenShift installation requires a pull secret obtained from try.openshift.com. You can specify an individual pull secret for each cluster Hive creates, or you can use a global pull secret that will be used by all of the clusters Hive creates.
 
 ```bash
-oc create secret generic global-pull-secret --from-file=.dockerconfigjson=<path>/config.json --type=kubernetes.io/dockerconfigjson --namespace hive
+oc create secret generic mycluster-pull-secret .dockerconfigjson=/path/to/pull-secret --type=kubernetes.io/dockerconfigjson --namespace hive
 ```
 
-Edit the HiveConfig to add global pull secret.
+```yaml
+apiVersion: v1
+data:
+  .dockerconfigjson: REDACTED
+kind: Secret
+metadata:
+  name: mycluster-pull-secret
+  namespace: hive
+type: kubernetes.io/dockerconfigjson
+```
+
+When a global pull secret is defined in the `hive` namespace and a `ClusterDeployment`-specific pull secret is specified, the registry authentication in both secrets will be merged and used by the new OpenShift cluster.
+When a registry exists in both pull secrets, precedence will be given to the contents of the cluster-specific pull secret.
+
+The global pull secret must live in the `hive` namespace and is referenced in the `HiveConfig` (also in the `hive` namespace).
+
+```yaml
+apiVersion: v1
+data:
+  .dockerconfigjson: REDACTED
+kind: Secret
+metadata:
+  name: global-pull-secret
+  namespace: hive
+type: kubernetes.io/dockerconfigjson
+
+```
 
 ```bash
 oc edit hiveconfig hive
 ```
 
-The global pull secret name must be configured in the HiveConfig CR.
-
 ```yaml
 spec:
   globalPullSecret:
     name: global-pull-secret
 ```
 
-### Provisioning
+### OpenShift Version
 
-Hive supports two methods of specifying what version of OpenShift you wish to install. Most commonly you can create a ClusterImageSet which references an OpenShift 4 release image.
+Hive needs to know what version of OpenShift to install. You can specify an individual OpenShift release image for each cluster Hive creates, or you can use a global `ClusterImageSet` that will be used by all of the clusters Hive creates.
 
-An example ClusterImageSet:
+An example `ClusterImageSet`:
 
 ```yaml
 apiVersion: hive.openshift.io/v1
 kind: ClusterImageSet
 metadata:
-  name: openshift-v4.2.0
+  name: openshift-v4.3.0
 spec:
-  releaseImage: quay.io/openshift-release-dev/ocp-release:4.2.0
+  releaseImage: quay.io/openshift-release-dev/ocp-release:4.3.0
+```
+Alternatively you can specify an OpenShift 4 release image directly on your `ClusterDeployment`. This value will override any value set in your `ClusterImageSet`.
```

Comment:
We should document where it's set, clusterDeployment.spec.provisioning.releaseImage.

Reply:
done